src/main/java/org/breedinginsight/api/v1/controller/brapi/BrAPIServiceFilter.java

```diff
@@ -0,0 +1,106 @@
+package org.breedinginsight.api.v1.controller.brapi;
+
+import io.micronaut.context.annotation.Value;
+import io.micronaut.core.order.Ordered;
+import io.micronaut.http.HttpAttributes;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.filter.OncePerRequestHttpServerFilter;
+import io.micronaut.http.filter.ServerFilterChain;
+import io.micronaut.web.router.MethodBasedRouteMatch;
+import io.micronaut.web.router.RouteMatch;
+import io.reactivex.Flowable;
+import io.reactivex.schedulers.Schedulers;
+import org.brapi.client.v2.model.exceptions.HttpInternalServerError;
+import org.brapi.client.v2.model.exceptions.HttpNotFoundException;
+import org.breedinginsight.services.brapi.BrAPIClientProvider;
+import org.breedinginsight.model.ProgramBrAPIEndpoints;
+import org.breedinginsight.services.ProgramService;
+import org.reactivestreams.Publisher;
+
+import javax.inject.Inject;
+import java.util.Map;
+import java.util.UUID;
+
+@Filter("/**/programs/**")
+public class BrAPIServiceFilter extends OncePerRequestHttpServerFilter {
+
+
+    @Value("${micronaut.brapi.server.url}")
+    private String defaultBrAPIUrl;
```

Comment:
Sorry I didn't catch this sooner, we will probably want to have a default BrAPI URL for each of the services (core, pheno, geno)